olp-cpp-sdk-dataservice-write/src/CatalogSettings.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+* Copyright (C) 2020 HERE Europe B.V.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+* License-Filename: LICENSE
+*/
+
+#include "CatalogSettings.h"
+
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/core/cache/CacheSettings.h>
+#include "ApiClientLookup.h"
+#include "generated/ConfigApi.h"
+
+// clang-format off
+#include <generated/serializer/CatalogSerializer.h>
+#include <generated/serializer/JsonSerializer.h>
+// clang-format on
+
+// clang-format off
+#include <generated/parser/CatalogParser.h>
+#include <olp/core/generated/parser/JsonParser.h>
+// clang-format on
+
+namespace olp {
+namespace dataservice {
+namespace write {
+
+CatalogSettings::CatalogSettings(const client::HRN catalog,
+                                 client::OlpClientSettings settings)
+    : catalog_(std::move(catalog)),
+      cache_(settings.cache),
+      settings_(std::move(settings)) {
+    if (!settings.cache) {
+        settings.cache = client::OlpClientSettingsFactory::CreateDefaultCache({});
+        cache_ = settings.cache;
+    }
+}
+
+CatalogSettings::LayerSettingsResult CatalogSettings::GetLayerSettings(client::CancellationContext context,
+                                                                  BillingTag billing_tag,
+                                                                  const std::string& layer_id) {
+  const auto catalog_settings_key = catalog_.ToString() + "::catalog";
```

Comment:
just curious, should we bother about catalog version?

Reply:
No need for version here. As this is global settings for the entire catalog.